include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -301,6 +301,12 @@
 #define _ONEDPL_CPP20_SHIFT_LEFT_RIGHT_PRESENT                                                                         \
     (_ONEDPL___cplusplus >= 202002L && (_MSC_VER >= 1921 || _GLIBCXX_RELEASE >= 10))
 
+#if defined(_MSVC_LANG)
+#define _ONEDPL_CPP20_RANGES_PRESENT (_MSVC_LANG >= 202002L)
+#else
+#define _ONEDPL_CPP20_RANGES_PRESENT (__cplusplus >= 202002L)
```

Comment:
Would basing this on the `__cpp_lib_ranges`  feature macro make sense?

We recently introduced the usage of feature macros in this header: https://github.com/oneapi-src/oneDPL/blob/acef1ad13f58345e4d502d801fc8dde3f0b6ee85/include/oneapi/dpl/pstl/onedpl_config.h#L304

Reply:
looks reasonable... But, we have to understand which value to check..
https://en.cppreference.com/w/cpp/feature_test
I guess, `__cpp_lib_ranges >= 202302L`

Reply:
It looks like `202302L` corresponds to a C++23 feature, so I think this is probably to high. It looks like the minimum is `201911L` so this may be sufficient unless we are using any of the features in the subsequent versions.